File: backend/components/metiers/threed/tests_centralises/api/samples/accessibility/test_sample_accessibility.js
// test_sample_accessibility.js – Test ultra avancé des samples accessibilité API Threed (centralisé)

const { sampleAccessibilityCheck } = require('../../../api/samples/accessibility');

describe('Samples Accessibilité API Threed', () => {
  it('doit valider l’accessibilité d’une ressource complète', () => {
    const data = { label: 'Ultra', lang: 'fr' };
    const result = sampleAccessibilityCheck(data);
    expect(result.accessible).toBe(true);
    expect(result.lang).toBe('fr');
  });

  it('doit détecter une ressource non accessible (label manquant)', () => {
    const data = { lang: 'fr' };
    const result = sampleAccessibilityCheck(data);
    expect(result.accessible).toBe(false);
    expect(result.reason).toBe('missing_label');
  });
});

File: backend/components/metiers/threed/tests_centralises/api/samples/accessibility/test_sample_accessibility.py
import unittest
from threed.api.samples.accessibility.sample_accessibility import (
    sample_accessibility_check,
)


class TestSampleAccessibility(unittest.TestCase):
    def test_accessibility_compliance(self):
        # Teste la conformité accessibilité sur un cas valide
        data = {"label": "Test", "lang": "fr"}
        result = sample_accessibility_check(data)
        self.assertTrue(result["accessible"])
        self.assertEqual(result["lang"], "fr")

    def test_accessibility_missing_label(self):
        # Teste le cas d'une ressource sans label
        data = {"lang": "fr"}
        result = sample_accessibility_check(data)
        self.assertFalse(result["accessible"])
        self.assertEqual(result["reason"], "missing_label")

    def test_accessibility_default_lang(self):
        # Teste la langue par défaut
        data = {"label": "Test"}
        result = sample_accessibility_check(data)
        self.assertEqual(result["lang"], "fr")
